source/ncengine/graphics2/diligent/resource/GlobalTextureBufferResource.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "Common/interface/RefCntAutoPtr.hpp"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/DeviceContext.h"
+
+#include <span>
+#include <vector>
+
+namespace nc
+{
+namespace asset
+{
+struct TextureWithId;
+} // namespace asset
+
+namespace graphics
+{
+class GlobalTextureBufferResource
+{
+    public:
+        explicit GlobalTextureBufferResource(Diligent::IShaderResourceVariable& variable, uint32_t maxTextures)
+            : m_variable{&variable},
+              m_maxTextures{maxTextures}
+        {
+        }
+
+        static auto MakeResourceDesc(std::string_view variableName, uint32_t maxTextures) -> Diligent::PipelineResourceDesc;
+        static auto MakeSamplerDesc(std::string_view variableName) -> Diligent::ImmutableSamplerDesc;
+
+        void Load(std::span<const asset::TextureWithId> textures,
+                  Diligent::IDeviceContext& context,
+                  Diligent::IRenderDevice& device);
+
+        void Unload();
+
+        auto GetShaderVariable() -> Diligent::IShaderResourceVariable&
+        {
+            return *m_variable;
+        }
+
+        auto GetMaxTextureCount() const -> size_t
+        {
+            return m_maxTextures;
+        }
+
+    private:
+        std::vector<Diligent::RefCntAutoPtr<Diligent::ITexture>> m_textures;
+        std::vector<Diligent::IDeviceObject*> m_objects;
```

Comment:
I'm curious about why m_objects is of type `IDeviceObject` rather than `ITextureView`? I'm not %100, but that seems to be the type we're returning here. If it can't be/isn't an `ITextureView` for some reason we should probably still call this member `m_views`

Reply:
`m_views` is a better name. The type is `IDeviceObject` because that's what the `IShaderResourceVariable` functions operate on. `ITextureView` is derived from it.